### PR TITLE
Trim whitespace for password #patch

### PR DESCRIPTION
**Pull request by stephen37**
Signed-off-by: stephen batifol <stephen.batifol@wolt.com>

# TL;DR
When deploying Flyte on our EKS cluster, we encountered a problem because we had a whitespace character, by trimming space, we should avoid that problem. Also it doesn't trim space if it is percent-encoded.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
Use the function [`TrimSpace`](https://pkg.go.dev/strings#TrimSpace) that "_returns a slice of the string s, with all leading and trailing white space removed, as defined by Unicode._"

## Tracking Issue
I had written something about it in a discussion here https://github.com/flyteorg/flyte/discussions/1472

### CODE REVIEW

File: pkg/runtime/application_config_provider.go [hunk 1]
```diff
@@ -38,7 +39,7 @@ func (p *ApplicationConfigurationProvider) GetDbConfig() dbconfig.DbConfig {
 			logger.Fatalf(context.Background(), "failed to read database password from path [%s] with err: %v",
 				dbConfigSection.PasswordPath, err)
 		}
-		password = string(passwordVal)
+		password = strings.TrimSpace(string(passwordVal))
```
Comment (eapolinario) [inline pkg/runtime/application_config_provider.go:42]: Can you link to https://www.postgresql.org/docs/current/libpq-connect.html and mention that passwords can contain special characters as long as they are percent-encoded?

Comment (stephen37) [reply pkg/runtime/application_config_provider.go:42]: Sure, do you want me to add it in file directly as a comment or somewhere else?

Comment (stephen37) [reply pkg/runtime/application_config_provider.go:44]: I added it as a comment.

**Review by eapolinario — APPROVED** [review]
